### PR TITLE
Feature/add dto layer

### PR DESCRIPTION
**Pull request by anishmu20**
Overview:

This PR introduces several enhancements and corrections to the library management system, particularly focusing on the API functionalities and code maintainability. Below are the detailed changes made:

1.Added Data Transfer Object (DTO) Layer:

Implemented a DTO layer for the module whose entity is exposed in the controller. This change facilitates data encapsulation and improves the separation of concerns.

2.Corrected Borrowing API (BorrowBook API):
Resolved an issue where the BorrowBook API was returning null for both book and member records. The API is now functioning correctly and provides the expected output.

3.Updated Return Book API:

The Return Book API was previously not returning any value. This has been corrected, and the return structure now aligns with the borrowing API documentation.

4.Lombok Dependency Implementation:
Integrated Lombok to reduce boilerplate code. Getter and setter methods, as well as constructors, have been generated using Lombok annotations, enhancing code readability and maintainability.

5.Introduced ModelMapper:
Implemented ModelMapper to facilitate seamless conversion between DTO and entity objects, ensuring cleaner and more efficient data handling.
Notification Table
members Table
Borrowings Table
Borrowings Table

6.API Testing:
Conducted thorough testing of all APIs using Postman to ensure that all functionalities are working as expected.

### CODE REVIEW

**Comment by anishmu20** [comment]
[Feature/add dto layer](https://github.com/ajaynegi45/LibraryMan-API/pull/61#top)
#52

File: pom.xml [hunk 1]
```diff
@@ -37,6 +37,17 @@
 			<artifactId>spring-boot-starter-data-jpa</artifactId>
 		</dependency>
 
+		<dependency>
+			<groupId>org.projectlombok</groupId>
+			<artifactId>lombok</artifactId>
```
Comment (ajaynegi45) [inline pom.xml:42]: Don't use lombok.

File: pom.xml [hunk 2]
```diff
@@ -37,6 +37,17 @@
...
+			<optional>true</optional>
+		</dependency>
+		<dependency>
+			<groupId>org.modelmapper</groupId>
+			<artifactId>modelmapper</artifactId>
```
Comment (ajaynegi45) [inline pom.xml:47]: Don't use modelmapper. Implement it manually

File: src/main/java/com/libraryman_api/book/Book.java [hunk 3]
```diff
@@ -33,91 +41,5 @@ public class Book {
     @Column(name = "copies_available", nullable = false)
     private int copiesAvailable;
 
-    public Book() {
```
Comment (ajaynegi45) [inline src/main/java/com/libraryman_api/book/Book.java:36]: Why you remove this code. Don't use lombok.

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -1,2 +1,3 @@
 spring.application.name=libraryman-api
```
Comment (ajaynegi45) [inline src/main/resources/application.properties:1]: Make this file as it was earlier.

File: src/main/resources/application-development.properties [hunk 5]
```diff
@@ -4,42 +4,45 @@
 ## Make this file as it was earlier before commiting code
 
 # Change this connection string to this format: jdbc:mysql://{ip_address}:{port}/{database_name}
-spring.datasource.url=jdbc:mysql://localhost:3306/Add_Your_Database_Name
+spring.datasource.url=jdbc:mysql://localhost:3306/DATABASE NAME
```
Comment (ajaynegi45) [inline src/main/resources/application-development.properties:7]: Make this file as it was earlier

**Review by ajaynegi45 — CHANGES_REQUESTED** [review]
Hi @ ,

Thank you for your prompt work on this issue. I appreciate your dedication to the project.

I have reviewed your work, and I need some changes.

- Check code comment and resolve them.
- Before making pull request always make sure to pull changes from upstream main branch.

Once again, thank you for your contribution! ❤️

**Pull request closed without merging**